### PR TITLE
Setup Docker & Contribution Guide

### PR DESCRIPTION
**Pull request by midtownkc**
Add Docker Compose & Contribution Guide

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+-   To work on something new, create a descriptively named branch off of main (ie: add-new-feature)
+-   Commit to the new branch locally and regularly push your work to the same named branch on GitHub
+-   If you need feedback or help, or you think a branch is ready to merge, open a pull request
+-   Remediate issues uncovered by GitHub Actions linters & automated testing
+-   Wait for code review & suggestions
+
+## Local Development Setup with Docker Compose
+
+1. [Install Docker Desktop](https://www.docker.com/products/docker-desktop/)
+1. Build Docker Images Locally: `docker compose build`
```
Comment (nat986) [inline CONTRIBUTING.md:25]: While attempting `docker compose build` on M1 MacBook Pro, docker build fails with error on step 3/7:
```
ERROR [bonne-bouche-app 3/7] RUN curl -sL https://dl.google.com/linux/linux_signing_key.pub > /etc/apt/trusted.gpg.d/google_linux_signing_key.asc     && echo "deb http://dl.google.com/linux/chrome/deb/ stable main" | tee /etc/apt/sources.list.d/google_chrome.list     && apt-get update -qq     && ap  2.6s
 => [bonne-bouche-css internal] load build context                                                                                                                                                                                                                                                               0.0s
 => => transferring context: 4.44kB                                                                                                                                                                                                                                                                              0.0s
 => [bonne-bouche-js internal] load build context                                                                                                                                                                                                                                                                0.0s
 => => transferring context: 4.44kB                                                                                                                                                                                                                                                                              0.0s
 => [bonne-bouche-web internal] load build context                                                                                                                                                                                                                                                               0.0s
 => => transferring context: 4.44kB                                                                                                                                                                                                                                                                              0.0s
------
 > [bonne-bouche-app 3/7] RUN curl -sL https://dl.google.com/linux/linux_signing_key.pub > /etc/apt/trusted.gpg.d/google_linux_signing_key.asc     && echo "deb http://dl.google.com/linux/chrome/deb/ stable main" | tee /etc/apt/sources.list.d/google_chrome.list     && apt-get update -qq     && apt-get install -y     google-chrome-stable     && apt-get clean     && rm -rf /var/lib/apt/lists/* /usr/share/doc /usr/share/man:
#0 0.308 deb http://dl.google.com/linux/chrome/deb/ stable main
#0 2.037 Reading package lists...
#0 2.323 Building dependency tree...
#0 2.404 Reading state information...
#0 2.458 E: Unable to locate package google-chrome-stable
------
failed to solve: executor failed running [/bin/sh -c curl -sL https://dl.google.com/linux/linux_signing_key.pub > /etc/apt/trusted.gpg.d/google_linux_signing_key.asc     && echo "deb http://dl.google.com/linux/chrome/deb/ stable main" | tee /etc/apt/sources.list.d/google_chrome.list     && apt-get update -qq     && apt-get install -y     google-chrome-stable     && apt-get clean     && rm -rf /var/lib/apt/lists/* /usr/share/doc /usr/share/man]: exit code: 100
```

**Review by nat986 — CHANGES_REQUESTED** [review]
Failed at step to docker compose build, see comment

**Comment by nat986** [comment]
Found documentation about development environment EVs: https://docs.docker.com/engine/reference/commandline/cli/#environment-variables

DOCKER_DEFAULT_PLATFORM can be set to indicate which platform images should default to, when multiple architectures are available

File: CONTRIBUTING.md (hunk 1 above)
Comment (midtownkc) [reply CONTRIBUTING.md:25]: Instructions are in the Required ENVs, will add to Docker Compose Setup as well

**Comment by nat986** [comment]
Docker build step now works. Running into another issue, now:

```
bonne-bouche % docker compose run web rails db:setup
[+] Running 2/0
 ⠿ Network bonne-bouche_default                                                                                                                            Created 0.0s
 ⠿ Container bonne-bouche-postgres-1                                                                                                                       Created 0.0s
 ⠋ postgres The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested  0.0s
[+] Running 1/1
 ⠿ Container bonne-bouche-postgres-1  Started                                                                                                              0.2s
Usage:
  rails new APP_PATH [options]

Options:
      [--skip-namespace], [--no-skip-namespace]              # Skip namespace (affects only isolated engines)
      [--skip-collision-check], [--no-skip-collision-check]  # Skip collision check
  -r, [--ruby=PATH]                                          # Path to the Ruby binary of your choice
                                                             # Default: /usr/local/bin/ruby
  -m, [--template=TEMPLATE]                                  # Path to some application template (can be a filesystem path or URL)
  -d, [--database=DATABASE]                                  # Preconfigure for selected database (options: mysql/postgresql/sqlite3/oracle/sqlserver/jdbcmysql/jdbcsqlite3/jdbcpostgresql/jdbc)
                                                             # Default: sqlite3
  -G, [--skip-git], [--no-skip-git]                          # Skip .gitignore file
      [--skip-keeps], [--no-skip-keeps]                      # Skip source control .keep files
  -M, [--skip-action-mailer], [--no-skip-action-mailer]      # Skip Action Mailer files
      [--skip-action-mailbox], [--no-skip-action-mailbox]    # Skip Action Mailbox gem
      [--skip-action-text], [--no-skip-action-text]          # Skip Action Text gem
  -O, [--skip-active-record], [--no-skip-active-record]      # Skip Active Record files
      [--skip-active-job], [--no-skip-active-job]            # Skip Active Job
      [--skip-active-storage], [--no-skip-active-storage]    # Skip Active Storage files
  -C, [--skip-action-cable], [--no-skip-action-cable]        # Skip Action Cable files
  -A, [--skip-asset-pipeline], [--no-skip-asset-pipeline]    # Indicates when to generate skip asset pipeline
  -a, [--asset-pipeline=ASSET_PIPELINE]                      # Choose your asset pipeline [options: sprockets (default), propshaft]
                                                             # Default: sprockets
  -J, [--skip-javascript], [--no-skip-javascript]            # Skip JavaScript files
      [--skip-hotwire], [--no-skip-hotwire]                  # Skip Hotwire integration
      [--skip-jbuilder], [--no-skip-jbuilder]                # Skip jbuilder gem
  -T, [--skip-test], [--no-skip-test]                        # Skip test files
      [--skip-system-test], [--no-skip-system-test]          # Skip system test files
      [--skip-bootsnap], [--no-skip-bootsnap]                # Skip bootsnap gem
      [--dev], [--no-dev]                                    # Set up the application with Gemfile pointing to your Rails checkout
      [--edge], [--no-edge]                                  # Set up the application with Gemfile pointing to Rails repository
  --master, [--main], [--no-main]                            # Set up the application with Gemfile pointing to Rails repository main branch
      [--rc=RC]                                              # Path to file containing extra configuration options for rails command
      [--no-rc], [--no-no-rc]                                # Skip loading of extra configuration options from .railsrc file
      [--api], [--no-api]                                    # Preconfigure smaller stack for API only apps
      [--minimal], [--no-minimal]                            # Preconfigure a minimal rails app
  -j, [--javascript=JAVASCRIPT]                              # Choose JavaScript approach [options: importmap (default), webpack, esbuild, rollup]
                                                             # Default: importmap
  -c, [--css=CSS]                                            # Choose CSS processor [options: tailwind, bootstrap, bulma, postcss, sass... check https://github.com/rails/cssbundling-rails]
  -B, [--skip-bundle], [--no-skip-bundle]                    # Don't run bundle install

Runtime options:
  -f, [--force]                    # Overwrite files that already exist
  -p, [--pretend], [--no-pretend]  # Run but do not make any changes
  -q, [--quiet], [--no-quiet]      # Suppress status output
  -s, [--skip], [--no-skip]        # Skip files that already exist

Rails options:
  -h, [--help], [--no-help]        # Show this help message and quit
  -v, [--version], [--no-version]  # Show Rails version number and quit

Description:
    The 'rails new' command creates a new Rails application with a default
    directory structure and configuration at the path you specify.

    You can specify extra command-line arguments to be used every time
    'rails new' runs in the .railsrc configuration file in your home directory,
    or in $XDG_CONFIG_HOME/rails/railsrc if XDG_CONFIG_HOME is set.

    Note that the arguments specified in the .railsrc file don't affect the
    defaults values shown above in this help message.

Example:
    rails new ~/Code/Ruby/weblog

    This generates a skeletal Rails installation in ~/Code/Ruby/weblog.
```

**Comment by midtownkc** [comment]
Forgot to add the WORKDIR to the Development Dockerfile.  rebuilding

**Comment by midtownkc** [comment]
It's all building locally again for me now & showing the default rails page when I do `docker compose up` and hit localhost:3000

**Review by nat986 — APPROVED** [review]
image

**Comment by isedwards** [comment]
[Banner](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
j42stin, @Fichte42, @Wostafa, @skozzii, @Shadow0ps, @ltbongo, @Code205, @ethereumdegen, @StanTowers, @thannaske, @RobertBohn, @qor-or-else, @MSylvia, @radegeddon, @schildbach, @Mekasan, @christinate, @ntorgov, @GPF, @Alexmike684, @RealCocoArdo, @theredrighthand, @rywards, @avelrl, @wlcdesigns, @nammba, @ckozus, @Cuda-Chen, @ftpmorph, @arthigo, @clittell, @i-love-to-code, @rdkamali, @erikson1970, @CamTosh, @Bila1x, @koalazak, @bitbug42, @itsallpretend, @skaefer143, @southpawgeek, @danielvigaru, @omega32, @harvysammen, @emre, @donttracemebruh, @ETDRFRS, @HashBiz, @intech, @975

**Pull request merged**